.github/workflows/repro-check.yml

```diff
@@ -0,0 +1,60 @@
+name: Repro check
+on:
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+    inputs:
+      github_sha:
+        description: Commit on which the repro check should be run
+        required: true
+        type: string
+  # Used as a reusable workflow within release-testing workflow
+  workflow_call:
+    inputs:
+      github_sha:
+        description: Commit on which the repro check should be run
+        required: true
+        type: string
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  repro-check:
+    name: Run repro-check for ${{ inputs.github_sha }}
+    runs-on:
+      labels: dind-large
+    container:
+      image: ubuntu:24.04
+      options: >-
+        -e NODE_NAME --privileged --cgroupns host
+    steps:
+      - name: Setup podman
+        shell: bash
+        run: |
+          apt-get update
+          apt-get -y upgrade
```

Comment:
Why this OS upgrade? Wouldn't that cause a lot of extra time and resources?

Reply:
It takes around 23-24 seconds. If we ever get down to this being a problem I think we are good to go but I can remove it.